components/mpas-ocean/src/mode_forward/mpas_ocn_time_integration_split_ab2.F

```diff
@@ -0,0 +1,3262 @@
+
+                     if (maxLevelEdgeTop(iEdge).eq.0) cycle
+
+                     flux = btrvel_temp(iEdge) * wctEdge(iEdge)
+
+                     sshTend(iCell) = sshTend(iCell) + &
+                                      edgeSignOnCell(i, iCell)* &
+                                      flux*dvEdge(iEdge)
+
+                  end do ! edges on cell
+
+                  ! SSHnew = SSHold + dt/J*(-div(Flux))
+                  sshSubcycleNew(iCell) = sshSubcycleCur(iCell) &
+                                     + dt/nBtrSubcycles * &
+                                       sshTend(iCell)/areaCell(iCell)
+               end do ! cell loop for ssh
+               !$omp end do
+               !$omp end parallel
+
+               !$omp parallel
+               !$omp do schedule(runtime)
+               do iCell = 1, nCells
+                  ! SSH is linear combination of SSHold and SSHnew
+                  sshTemp(iCell) = (1-config_btr_gam2_SSHWt1)* &
+                                   sshSubcycleCur(iCell) + &
+                                   config_btr_gam2_SSHWt1 * &
+                                   sshSubcycleNew(iCell)
+               end do ! cell loop for ssh
+               !$omp end do
+               !$omp end parallel
+
+               call ocn_diagnostic_solve_wctEdge(btrvel_temp, sshTemp, &
+                      bottomDepthEdge, wctEdge)
+
+               ! compute barotropic thickness flux on edges
+               !$omp parallel
+               !$omp do schedule(runtime) &
+               !$omp private(flux)
+               do iEdge = 1, nEdges
+
+                  flux = btrvel_temp(iEdge) * wctEdge(iEdge)
+
+                  barotropicThicknessFlux(iEdge) = &
+                  barotropicThicknessFlux(iEdge) + flux
+
+               end do ! edge barotropic thickness flux
+               !$omp end do
+               !$omp end parallel
+
+               edgeHaloComputeCounter = config_num_halos + 1
+
+               deallocate(sshTemp)
+
+            endif ! config_btr_solve_SSH2
+
+            !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+            ! Barotropic subcycle: Accumulate running sums, advance
+            !                      timestep pointers
+            !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+            call mpas_pool_get_array(statePool, &
+                               'normalBarotropicVelocitySubcycle', &
+                                normalBarotropicVelocitySubcycleNew, &
+                                newBtrSubcycleTime)
+
+            ! normalBarotropicVelocityNew =
+            ! normalBarotropicVelocityNew +
+            ! normalBarotropicVelocitySubcycleNEW
+            ! This accumulates the sum. If the Barotropic Coriolis
+            ! iteration is limited to one, this could
+            ! be merged with the above code.
+
+            !$omp parallel
+            !$omp do schedule(runtime)
+            do iEdge = 1, nEdgesAll
```

Comment:
@hyungyukang, per the bug fix from @cbegeman, this should be `nEdgesOwned`. I'm testing now. The partition tests will tell us if this is correct.

Reply:
@cbegeman and @mark-petersen , Thanks a lot. Please let me know once your test is done. And I will also test the code on larger meshes.

Reply:
I tested this loop with `nEdgesOwned` and it changes the solution, as expected from results in https://github.com/E3SM-Project/E3SM/pull/6041#pullrequestreview-1710892492. But for AB2, if I change both loops (lines 1542 and 1570) from `nEdgesOwned` to `nEdgesAll` the solution does not match for all tests running `split_explicit_ab2`. Since `nEdgesAll` is the more conservative approach, I am changing the loop extents to the larger option.

Reply:
@mark-petersen and @hyungyukang Thanks for following up on this! Glad to hear that the AB2 tests are going well.